### ISSUE TITLE
Allow for localized CLS styles

### ISSUE THREAD

**Issue by Fiech** [asciidoctor/asciidoctor-bibtex#34]
_Background: I want to localize the automated labels, e.g. "Ed." for the editor to the German "Hrsg."._

It may be a misunderstanding on my part, but per the Readme:

Attribute Name | Description | Valid Values | Default Value
-- | -- | -- | --
| bibtex-style | Reference formatting style | any style supported by csl-styles | ieee |

I had a look at the CLS website, particularly the _Localization_ section: https://citationstyles.org/authors/#/localization

In the linked page, there are a lot of localization options listed: https://github.com/citation-style-language/locales/wiki

Altough, when I'm trying to use one of the options, e,g. and i.e. "de-DE", I get the following error, when I'm trying to compile via `asciidoctor -r asciidoctor-bibtex test.adoc`:

> asciidoctor: FAILED: *REDACTED*/test.adoc: Failed to load AsciiDoc document - failed to extract CSL data from "de-DE": No such file or directory @ rb_sysopen - de-DE

My minimum NOT working example:

**File test.adoc:**

```asciidoc
= Title
:bibtex-file: bibliography.bib
:bibtex-style: de-DE

This is a sample text cite:[cite_id].

bibliography::[]
```

**File bibliography.bib:**

```
@book{cite_id,
   year={2018},
   title={Book Title},
   editor={John Editor},
}
```

Am I misinterpreting something?

**Comment by ProgramFan** [comment]
`bibtex-style` is the CSL **style**, not the locale. So setting it to `de-DE` would not work. But I have currently no idea on how to make CSL produce localized items. Any ideas on that?

**Comment by Fiech** [comment]
Thank you for clearing this up.

So, the styles have to be one of those in the [list](https://github.com/citation-style-language/styles ), or respectively one found via the [search function](https://github.com/citation-style-language/styles).

So, another parameter, e.g. `bibtex-locale` is needed, we can set to the ones listed [here](https://github.com/citation-style-language/locales/wiki).

Maybe the [bibtex-ruby gem](https://github.com/inukshuk/bibtex-ruby) already includes the possibility to set the locale? I'll have a look later.

**Comment by Fiech** [comment]
So, after some short digging around, I found this code snipped on [this web page](https://www.ramonvoges.de/werkzeugkasten/Bibtex-CSL-und-Ruby/) (sorry, it's in German):

```ruby
cp = CiteProc::Processor.new style: 'foerster-geisteswissenschaft', format: 'text', locale: 'de-DE'
```

A quick cross search in the `asciidoctor-bibtex` project yielded [this code](https://github.com/asciidoctor/asciidoctor-bibtex/blob/feed91ecd1130b80ed2820aabe92ba08101bbf62/lib/asciidoctor-bibtex/processor.rb#L27) as a possible vector for improvement:

```ruby
@citeproc = CiteProc::Processor.new style: @style, format: :html
```

At a later time, I can try testing if a hard-coded locale field yields the desired result.

**Comment by Fiech** [comment]
@ProgramFan Hey, I added a PR #35 that would introduce such a configurable locale.

I tested it succesfully with asciidoctor and asciidoctor-pdf

**Comment by Fiech** [comment]
BTW: the parameter is called `bibtex-locale`, so the test file becomes:

**test.adoc:**

```asciidoc
= Title
:bibtex-file: bibliography.bib
:bibtex-locale: de-DE

This is a sample text cite:[cite_id].

bibliography::[]
```

**Comment by Fiech** [comment]
This Issue should be closed with PR #35
